File: backend/test_observer/controllers/permissions/permissions.py
# Copyright 2025 Canonical Ltd.
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU Affero General Public License version 3, as
# published by the Free Software Foundation.
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU Affero General Public License for more details.
# You should have received a copy of the GNU Affero General Public License
# along with this program.  If not, see <https://www.gnu.org/licenses/>.
#
# SPDX-FileCopyrightText: Copyright 2025 Canonical Ltd.
# SPDX-License-Identifier: AGPL-3.0-only

from fastapi import APIRouter, Security

from test_observer.common.permissions import Permission, permission_checker

router: APIRouter = APIRouter(tags=["permissions"])


@router.get("", dependencies=[Security(permission_checker, scopes=[Permission.view_permission])])
def get_permissions():
    return [p.value for p in Permission]

File: frontend-svelte/src/lib/api/permissions.ts
// SPDX-FileCopyrightText: Copyright 2024 Canonical Ltd.
// SPDX-License-Identifier: GPL-3.0-only

import { apiFetch } from "./client.js";

export function getPermissions(
  fetchFn: typeof fetch = fetch,
): Promise<string[]> {
  return apiFetch<string[]>("/v1/permissions", {}, fetchFn);
}
